### PR TITLE
Add authentication_status attribute to NotificationSource

### PR DESCRIPTION
**Pull request by chadell**
Add a validate authentication for notification sources to give a quick understanding of it works or not before actually fetching notifications.
My goal behind this PR is to facilitate adding new `NotificationSources` as pointed out in issue #15  which I will start after this PR.

image

### CODE REVIEW

**Review by glennmatthews — COMMENTED** [review]
General comment/question - does it actually make sense to store authentication status as a DB field? Or is it something that should be dynamically generated on request (like a device's LLDP neighbors) since it isn't really something that persists -- authentication may succeed **at this moment** but silently change to failing later because the account's credentials were changed or such?

**Comment by chadell** [comment]
> General comment/question - does it actually make sense to store authentication status as a DB field? Or is it something that should be dynamically generated on request (like a device's LLDP neighbors) since it isn't really something that persists -- authentication may succeed **at this moment** but silently change to failing later because the account's credentials were changed or such?

This is actually a good point... I implemented this because I was foresee that to use Oauth later I will need an initial login from the user to generate the first token and keep refreshing... so I would force the operator to login it at least once.
But this won't actually work between application resets... so I believe that I will have to use Google `Service Accounts`...
and this rational is to conclude that you are right, we can't assume that what was authenticated in the past is it still  in the present.

So I will keep the feature but without actually storing it, just as a dynamic data

File: nautobot_circuit_maintenance/handle_notifications/sources.py [hunk 1]
```diff
@@ -117,6 +127,27 @@ def close_session(self):
...
+        """
+        try:
+            self.open_session()
+            self.close_session()
+            is_authenticated = True
+            message = "Test OK"
+        except Exception as exc:
+            is_authenticated = False
+            if isinstance(exc.args[0], bytes):
+                message = str(exc.args[0].decode())
```
Comment (glennmatthews) [inline nautobot_circuit_maintenance/handle_notifications/sources.py:146]: What's the scenario that this `if` condition is handling? I would have thought `str(exc)` would generally be the desired output?

**Review by glennmatthews — APPROVED** [review]
Looks good!

File: nautobot_circuit_maintenance/handle_notifications/sources.py (hunk 1 above)
Comment (chadell) [reply nautobot_circuit_maintenance/handle_notifications/sources.py:146]: I was getting a `str` from `bytes` such as `b'ABC'`, so `decoding` first removes the `b''` (maybe there is a better way :) )

**Pull request merged**